test/tests/eclipse-mosquitto-basics/run.sh

```diff
@@ -15,11 +15,13 @@ dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
 
 image="$1"
 
-cname="eclipse-mosquitto-container-$RANDOM-$RANDOM"
-cid="$(docker run -d \
-	--name "$cname" \
-	"$image"
-)"
+# Create an instance of the container-under-test
+cname="$("$dir/../image-name.sh" librarytest/eclipse-mosquitto-basics "$image")"
+"$dir/../docker-build.sh" "$dir" "$cname" <<EOD
+FROM $image
+COPY dir/mosquitto.conf /mosquitto/config/
+EOD
+cid="$(docker run -d "$cname")"
```

Comment:
Try something like this:

```suggestion
serverImage="$("$dir/../image-name.sh" librarytest/eclipse-mosquitto-basics "$image")"
"$dir/../docker-build.sh" "$dir" "$serverImage" <<EOD
FROM $image
COPY dir/mosquitto.conf /mosquitto/config/
EOD
cname="eclipse-mosquitto-container-$RANDOM-$RANDOM"
cid="$(docker run -d \
	--name "$cname" \
	"$serverImage"
)"
```

Reply:
Thank you, I think I got there in the end. Sorry for the noise, I should've left it for the morning after the first fail.